odoo/fields.py

```diff
@@ -1550,10 +1554,32 @@ def _setup_attrs(self, model_class, name):
             self.prefetch = not callable(self.translate)
 
     _related_translate = property(attrgetter('translate'))
+    _related_pattern = property(attrgetter('pattern'))
 
     def _description_translate(self, env):
         return bool(self.translate)
 
+    def _description_pattern(self, env):
+        if self.pattern[0]:
+            string, pattern, pyflag = self.pattern
+            if env.lang:
+                model_name = self.base_field.model_name
+                field_pattern = env['ir.translation'].get_field_pattern(model_name)
+                string = field_pattern.get(self.name) or string
+            flag = ''.join(
+                js_flag
+                for py_flag, js_flag in (
+                    (re.I, "i"),
+                    (re.M, "m"),
+                    (re.S, "s"),
```

Comment:
You should move that mapping to a module constant.
Or alternatively define a conversion function in the module (this would help maintenance).